source/system/system.go

```diff
@@ -101,6 +102,15 @@ func (s *systemSource) Discover() error {
 		}
 	}
 
+	// Get vendor information
+	sys_vendor, err := getDmiIdAttribute("sys_vendor")
+	if err != nil {
+		klog.ErrorS(err, "failed to get sys_vendor")
+	} else {
+		s.features.Attributes[DmiIdFeature] = nfdv1alpha1.NewAttributeFeatures(nil)
+		s.features.Attributes[DmiIdFeature].Elements["sys_vendor"] = sys_vendor
+	}
```

Comment:
Now that we're on this, we could make it a bit more generic so that adding new dmi-id attributes in the future is super simple (with little risk of breaking anything existing).

For example something like this (or specifying one additional helper func to do this):

```suggestion
	dmiIdAttributeNames := []string{"sys_vendor"}
	dmiAttrs := map[string]string
	for name := range dmiIdAttributeNames {
		if val, err := getDmiIdAttribute(name); err {
			klog.ErrorS(err, "failed to get DMI entry", "attributeName", name)
		} else {
			dmiAttrs[name] = val
		}
	}
	s.features.Attributes[DmiIdFeature] = nfdv1alpha1.NewAttributeFeatures(dmiAttrs)
```

WDYT?

Reply:
@marquiz
Thanks, makes sense, updated.